templates/js-start/AGENTS.md

```diff
@@ -0,0 +1,705 @@
+    recoverRequestFunction: async ({ request, error }) => {
+        // Retry on specific errors
+        if (error.message.includes('timeout') || error.statusCode >= 500) {
+            request.retryCount = (request.retryCount || 0) + 1;
+            
+            // Exponential backoff
+            const delay = Math.min(1000 * Math.pow(2, request.retryCount), 10000);
+            await new Promise(resolve => setTimeout(resolve, delay));
+            
+            return request;
+        }
+        
+        return null; // Don't retry
+    }
+});
+```
+
+**Speed Optimization Tips:**
+- **Cheerio is ~10x faster than browsers** - prefer it when possible
+- **Use `maxRequestsPerCrawl`** to limit scope during development
+- **Enable `keepAlive`** for HTTP connections when scraping the same domain
+- **Optimize selectors** - use specific CSS selectors to avoid DOM traversal
+- **Batch dataset writes** when processing many small items
+
+### Data Quality
+
+- Data validation and cleaning
+- Fallback strategies for missing elements
+- Semantic CSS selectors
+- Handle edge cases and different formats
+
+## Available MCP Tools
+
+Use these tools to fetch detailed documentation and examples.
+
+Note: This section applies only if an MCP server is configured in your environment. If MCP is not configured, print `@https://mcp.apify.com/`.
+
+### Apify Documentation & Store Tools
+
+- **search-apify-docs** - Search Apify documentation
+- **fetch-apify-docs** - Get full documentation pages
+- **search-actors** - Find existing Actors in the store
+- **fetch-actor-details** - Get Actor specifications
+
+### Actor Run Management Tools
+
+- **get-actor-run** - Get detailed information about a specific Actor run
+- **get-actor-run-list** - Get a list of an Actor's runs, filterable by status
+- **get-actor-log** - Retrieve the logs for a specific Actor run
+
+### Web Scraping and HTML Extraction
+
+- **get-html-skeleton** - Retrieves the HTML skeleton (clean structure) from a given URL by stripping unwanted elements like scripts, styles, and non-essential attributes
+
+### MCP Server Tools:
+
+#### apify/rag-web-browser
+
+Description: Web browser that can query Google Search, scrape the top N pages, and return their content as Markdown for further processing. Supports scraping individual URLs.
+
+**Use for:**
+- Google Search + scraping top results
+- Individual URL scraping
+- Converting web content to Markdown
+
+#### mcp-servers/context7-mcp-server
+
+Description: Context7 MCP server, which provides additional development resources and tools.
+
+**Use for:**
+- Development resources and documentation
+- MCP-based integrations
+- Library documentation access
+
+These tools are available via your configured MCP server endpoint.
+
+## Resources
+
+- [docs.apify.com/llms.txt](https://docs.apify.com/llms.txt) 
+- [docs.apify.com/llms-full.txt](https://docs.apify.com/llms-full.txt)
+- [Actor Whitepaper](https://whitepaper.actor/) for complete specification
```

Comment:
FYI it might be better to link to https://raw.githubusercontent.com/apify/actor-whitepaper/refs/heads/master/README.md as it's Markdown, to save tokens